src/pytorch_lightning/tuner/lr_finder.py

```diff
@@ -380,7 +388,7 @@ def on_train_batch_end(
         # Check if we diverging
         if self.early_stop_threshold is not None:
             if current_step > 1 and smoothed_loss > self.early_stop_threshold * self.best_loss:
-                trainer.fit_loop.max_steps = current_step  # stop signal
+                trainer.should_stop = True  # stop signal
```

Comment:
We need some mechanism that all ranks make this decision simultaneously. Either guarantee that the if condition is always the same in all ranks, or reduce the boolean decision across all ranks. I couldn't see where we are doing this. See also #15253

Reply:
good point!
since we rely on lr of rank zero.. let's broadcast?
I am not sure what's the best way here. If we do all reduce, then it's not good to just broadcast the lr at the end of lr_Find, we should reduce that well too.

Reply:
Yean, I think doing a broadcast here is definitely one solution you could do. For now, that should be ok. The tuner is anyway only a heuristic, we don't need to be super sophisticated here with the stopping criterion. 


FYI:

> If we do all reduce, then it's not good to just broadcast the lr at the end of lr_Find, we should reduce that well too.

That wouldn't be the same. Reducing the boolean decision here means either
- set stop True if any of the ranks want to stop
- set stop True if all of the ranks want to stop

reduce_boolean_decision does the latter, but after https://github.com/Lightning-AI/lightning/pull/15253 we could have the former too.

Reply:
then let's do broadcast when any of the ranks stop and broadcast it's learning rate